independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/CreateExtension.java

```diff
@@ -0,0 +1,455 @@
+package io.quarkus.devtools.commands;
+
+import static io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData.*;
+import static io.quarkus.devtools.commands.handlers.CreateExtensionCommandHandler.readPom;
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+import io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog;
+import io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData;
+import io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartProjectInput;
+import io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartProjectInputBuilder;
+import io.quarkus.devtools.commands.data.QuarkusCommandException;
+import io.quarkus.devtools.commands.data.QuarkusCommandOutcome;
+import io.quarkus.devtools.commands.handlers.CreateExtensionCommandHandler;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+import io.quarkus.maven.utilities.MojoUtils;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Stack;
+import java.util.stream.Collectors;
+import javax.lang.model.SourceVersion;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.model.Model;
+
+/**
+ * Instances of this class are not thread-safe. They are created per invocation.
+ */
+public class CreateExtension {
+
+    public enum LayoutType {
+        OTHER_PLATFORM,
+        QUARKUS_CORE,
+        QUARKIVERSE,
+        STANDALONE
+    }
+
+    public static final String DEFAULT_BOM_GROUP_ID = "io.quarkus";
+    public static final String DEFAULT_BOM_ARTIFACT_ID = "quarkus-bom";
+    public static final String DEFAULT_BOM_VERSION = "${quarkus.version}";
+    public static final String DEFAULT_VERSION = "0.0.1-SNAPSHOT";
+
+    public static final String DEFAULT_CORE_NAMESPACE_ID = "quarkus-";
+
+    public static final String DEFAULT_EXTERNAL_NAMESPACE_ID = "";
+
+    public static final String DEFAULT_QUARKIVERSE_PARENT_GROUP_ID = "io.quarkiverse";
+    public static final String DEFAULT_QUARKIVERSE_PARENT_ARTIFACT_ID = "quarkiverse-parent";
+    public static final String DEFAULT_QUARKIVERSE_PARENT_VERSION = "5";
```

Comment:
Hint: these properties can be moved as getters inside the `LayoutType` enum above

Reply:
Good hint 😇